backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/callout/C_OrderLine.java

```diff
@@ -47,6 +55,17 @@ public void updateQtyCU(final I_C_OrderLine orderLine, final ICalloutField field
 		Services.get(IOrderLineBL.class).updateLineNetAmtFromQtyEntered(orderLine);
 	}
 
+	@CalloutMethod(columnNames = de.metas.interfaces.I_C_OrderLine.COLUMNNAME_M_Product_ID)
+	public void onProductSetOrChanged(final de.metas.interfaces.I_C_OrderLine orderLine)
+	{
+		final Optional<I_M_HU_PI_Item_Product> huPiItemProduct = hupiItemProductDAO.retrieveDefaultForProduct(ProductId.ofRepoId(orderLine.getM_Product_ID()),
```

Comment:
* 1 i think it would be better to do this in a ModelInterceptor..i believe it make sense to do this also if e.g. the order line is created via scheduler from an C_OLCand
* 2 if the new product has no M_HU_PI_Item_Product_ID, then i think C_OrderLine.M_HU_PI_Item_Product_ID should be set to -1
* 3 please add a method `Optional<HUPIItemProductId> retrieveDefaultIdForProduct` to IHUPIItemProductDAO`
  * rationale: try to avoid throwing around data records. we like to grow in the general direction of DDD.
  * also, we don't need (to load) the record for this

Reply:
1. Moved to ModelInterceptor
2. HUPIItemProductId.toRepoId(null) will return -1, used that.
3. Added, though I've only used it for our particular case. There are other places which still call `retrieveDefaultForProduct`, but we'd need to test those changes....